# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/ioctl-prototype.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

The ioctl requeset is long (64bit on 64bit machines) and unsigned,
the compiler must not sign extend the value (that bites).

  - Rene Rebe <rene@exactcode.de>

diff -ur ../dietlibc-0.29-32/include/sys/ioctl.h ./include/sys/ioctl.h
--- ./include/sys/ioctl.h	2005-03-07 09:07:40.000000000 +0000
+++ ./include/sys/ioctl.h	2006-06-18 10:20:16.180000000 +0000
@@ -29,7 +29,7 @@
 #define EP_FREE		_IO('P', 3)
 #define EP_ISPOLLED	_IOWR('P', 4, struct pollfd)
 
-int ioctl(int d, int request, ...) __THROW;
+int ioctl(int d, unsigned long request, ...) __THROW;
 
 __END_DECLS
 
